src/capture/capture.cpp

```diff
@@ -290,110 +345,236 @@ static void handle_capture_midi_event(bool pressed)
+	case CaptureType::Midi: capture.next_index.midi = index; break;
+
+	case CaptureType::RawOplStream:
+		capture.next_index.raw_opl_stream = index;
+		break;
+
+	case CaptureType::RadOplInstruments:
+		capture.next_index.rad_opl_instrument = index;
+		break;
+
+	case CaptureType::Video: capture.next_index.video = index; break;
+
+	case CaptureType::RawImage:
+	case CaptureType::UpscaledImage:
+	case CaptureType::RenderedImage:
+		capture.next_index.image = index;
+		break;
+
+	case CaptureType::SerialLog:
+		capture.next_index.serial_log = index;
+		break;
+
+	default: assertm(false, "Unknown CaptureType");
 	}
-#endif
 }
 
-// TODO move raw OPL capture and serial log capture here too
+static bool create_capture_directory()
+{
+	std::error_code ec = {};
+
+	if (!std_fs::exists(capture.path, ec)) {
+		if (!std_fs::create_directory(capture.path, ec)) {
+			LOG_WARNING("CAPTURE: Can't create directory '%s': %s",
+			            capture.path.string().c_str(),
+			            ec.message().c_str());
+			return false;
+		}
+		std_fs::permissions(capture.path, std_fs::perms::owner_all, ec);
+	}
+	return true;
+}
 
-void CAPTURE_Init(Section* sec)
+static void capture_init(Section* sec)
 {
 	assert(sec);
+	const Section_prop* secprop = dynamic_cast<Section_prop*>(sec);
+	if (!secprop) {
+		return;
+	}
 
-	const Section_prop* conf = dynamic_cast<Section_prop*>(sec);
-	assert(conf);
+	Prop_path* capture_path = secprop->Get_path("capture_dir");
+	assert(capture_path);
+
+	// We can safely change the capture output path even if capturing of any
+	// type is in progress.
+	capture.path = std_fs::path(capture_path->realpath);
+	if (capture.path.empty()) {
+		LOG_WARNING("CAPTURE: No value specified for `capture_dir`; defaulting to 'capture' "
+		            "in the current working directory");
+		capture.path = "capture";
+	}
+
+	const std::string prefs = secprop->Get_string("default_image_capture_formats");
 
-	Prop_path* proppath = conf->Get_path("captures");
-	assert(proppath);
+	image_capturer = std::make_unique<ImageCapturer>(prefs);
 
-	capture_dir = proppath->realpath;
+	constexpr auto changeable_at_runtime = true;
+	sec->AddDestroyFunction(&capture_destroy, changeable_at_runtime);
+
+	if (!create_capture_directory()) {
+		return;
+	}
 
-	capturing_audio = false;
-	capturing_image = false;
-	capturing_midi  = false;
-	capturing_opl   = false;
-	capturing_video = false;
+	const std::vector<CaptureType> all_capture_types = {
+	        CaptureType::Audio,
+	        CaptureType::Midi,
+	        CaptureType::RawOplStream,
+	        CaptureType::RadOplInstruments,
+	        CaptureType::Video,
+	        CaptureType::RawImage,
+	        CaptureType::UpscaledImage,
+	        CaptureType::RenderedImage,
+	        CaptureType::SerialLog};
+
+	for (auto type : all_capture_types) {
+		const auto index = find_highest_capture_index(type);
+		if (index) {
+			set_next_capture_index(type, *index + 1);
+		} else {
+			return;
+		}
```

Comment:
If the first capture type returns zero, say audio, then we will quit scanning, and the other capture types' indices remain at default (or zero)?

Reply:
Hmm, better to start from 1, actually. Changing the defaults, good pickup.